substra/sdk/rest_client.py

```diff
@@ -115,6 +115,11 @@ def __request(self, request_name, url, **request_kwargs):
         kwargs = dict(self._default_kwargs)
         kwargs.update(request_kwargs)
 
+        # rewind files so that they are properly sent in retries as well
+        if 'files' in kwargs:
+            for _, file in kwargs['files'].items():
```

Comment:
```suggestion
            for file in kwargs['files'].values():
```

Really minor suggestion.

Reply:
I never think of iterating on values only, good idea!